packages/vue-app/template/mixins/fetch.client.js

```diff
@@ -64,7 +64,14 @@ function createdFullStatic() {
     return
   }
   this._hydrated = true
-  this._fetchKey = this.<%= globals.nuxt %>._payloadFetchIndex++
+
+  if (typeof this.$options.fetchKey === 'function') {
+    this._fetchKey = this.$options.fetchKey.call(this)
```

Comment:
We might still increase and pass `_payloadFetchIndex` as second argument to distinguish multiple instances of one component. (or `getKey` as second arg)

Reply:
If we want to support defaulting to the component name as a fetch key, as in linked issue, we will need a nested global fetchKey object to keep track of individual indices of each key. That is, the fetch state might look like:
```js
{
  '0': { a: 'mystate' },
  'MyComponent': { a: 'mystate' },
  'MyComponent': [{ a: 'mystate' }, { a: 'mystate2' }],
  '1': { a: 'mystate' },
}
```

If we don't keep track of these indices separately, the same issue that triggered the original bug will reappear because the global numeric fetch key will get out of sync.

Reply:
That's true. We might use a counter table for `getKey` implementation using component id as key..

Reply:
What would `getKey` API look like? Would it just return a valid incremented number unique to the string passed in?
```js
getKey (keyToLookup: string): number
```

Reply:
Ideally `getKey (id: string = ThisConstructor.id): number` but by default indeed using internal table to get a name spaced incrementor.